paddle/fluid/operators/mkldnn/mul_mkldnn_op.cc

```diff
@@ -0,0 +1,518 @@
+                                                y_desc, dst_desc);
+
+    return inner_product_forward::primitive_desc(mul_desc, engine_);
+  }
+
+  memory CreateDstMemory(
+      const inner_product_forward::primitive_desc &mul_prim_desc,
+      const ExecutionContext &ctx, Tensor *output) {
+    auto dst_prim_desc = mul_prim_desc.dst_primitive_desc();
+    auto buffer_size = dst_prim_desc.get_size();
+
+    float *output_data =
+        output->mutable_data<float>(ctx.GetPlace(), buffer_size);
+    output->set_format((memory::format)dst_prim_desc.desc().data.format);
+    return memory(dst_prim_desc, to_void_cast<float>(output_data));
+  }
+
+ protected:
+  const mkldnn::engine &engine_;
+  boost::optional<memory> x_input_;
+  boost::optional<memory> y_input_;
+  boost::optional<memory> output_;
+  boost::optional<inner_product_forward> mul_;
+};  // namespace operators
+
+template <typename T, typename K>
+class QuantMulPrimitiveFactory : public MulPrimitiveFactory<T, K> {
+ public:
+  using MulPrimitiveFactory<T, K>::MulPrimitiveFactory;
+
+  virtual inner_product_forward CreateMulPrimitive(
+      const Tensor *x_input, const Tensor *y_input, Tensor *output,
+      const ExecutionContext &ctx) {
+    /* check data format and reorder if need */
+    int x_num_col_dims = ctx.Attr<int>("x_num_col_dims");
+    int y_num_col_dims = ctx.Attr<int>("y_num_col_dims");
+    auto scale_y = ctx.Attr<std::vector<float>>("scale_y");
+
+    memory::data_type x_dt = framework::ToMKLDNNDataType(x_input->type());
+    memory::data_type y_dt = framework::ToMKLDNNDataType(y_input->type());
+    auto x_matrix = this->UpdateDataFormat(x_input, false, /*is_y*/
+                                           x_num_col_dims, x_dt, ctx);
+    auto y_matrix = this->UpdateDataFormat(y_input, true, /*is_y*/
+                                           y_num_col_dims, y_dt, ctx);
+
+    auto output_dim = output->dims();
+    if (output_dim.size() != 2) {
+      output->Resize({x_matrix.dims()[0], y_matrix.dims()[1]});
+    }
+
+    if (this->mul_) {
+      UpdateDataPointers(ctx, output, &x_matrix);
+      return *(this->mul_);
+    }
+
+    auto src_desc =
+        this->CreateMemDescriptor(&x_matrix, x_dt, memory::format::nc);
+    this->x_input_ = this->CreateMemory(src_desc, &x_matrix, false /*is_y*/);
+
+    if (std::is_same<K, float>::value) {
+      const auto trans_y_ = TransposeInputY(&y_matrix, y_dt);
+      this->y_input_ = QuantInputY(trans_y_, scale_y);
+    } else if (std::is_same<K, int8_t>::value) {
+      this->y_input_ = this->TransposeInputY(&y_matrix, y_dt);
+    } /* else { something goes wrong } */
+
+    bool force_fp32_output = ctx.Attr<bool>("force_fp32_output");
+    memory::data_type dst_dt;
+    if (force_fp32_output) {
+      dst_dt = framework::ToMKLDNNDataType(
+          framework::DataTypeTrait<float>::DataType);
+    } else {
+      dst_dt = framework::ToMKLDNNDataType(
+          framework::DataTypeTrait<int8_t>::DataType);
+    }
```

Comment:
This can be simplified to:
```
memory::data_type dst_dt = force_fp32_output ? MKLDNNGetDataType<float>() : MKLDNNGetDataType<int8_t>();
```
I think you'd need to add an appropriate overload in `mkldnn_helper.h` for `int8_t` type though, since its not yet available.